pages/instructorIssues/instructorIssues.js

```diff
@@ -1,17 +1,91 @@
-var ERR = require('async-stacktrace');
-var _ = require('lodash');
-var express = require('express');
-var router = express.Router();
+const ERR = require('async-stacktrace');
+const _ = require('lodash');
+const moment = require('moment');
+const express = require('express');
+const router = express.Router();
+const SearchString = require('search-string');
 
-var error = require('../../lib/error');
-var paginate = require('../../lib/paginate');
-var sqldb = require('../../lib/sqldb');
-var sqlLoader = require('../../lib/sql-loader');
+const error = require('../../lib/error');
+const paginate = require('../../lib/paginate');
+const sqldb = require('../../lib/sqldb');
+const sqlLoader = require('../../lib/sql-loader');
 
-var sql = sqlLoader.loadSqlEquiv(__filename);
+const sql = sqlLoader.loadSqlEquiv(__filename);
 
 const pageSize = 100;
 
+const commonQueries = {
+    allOpenQuery: 'is:open',
+    allClosedQuery: 'is:closed',
+    allManuallyReportedQuery: 'is:manually-reported',
+};
+
+const formattedCommonQueries = {};
+
+Object.keys(commonQueries).forEach((key) => {
+    formattedCommonQueries[key] = `?q=${encodeURIComponent(commonQueries[key])}`;
+});
+
+function formatForLikeClause(str) {
+    return `%${str}%`;
+}
+
+function parseRawQuery(str) {
+    const parsedQuery = SearchString.parse(str);
+    const filters = {
+        filter_is_open: null,
+        filter_is_closed: null,
+        filter_manually_reported: null,
+        filter_qids: null,
+        filter_not_qids: null,
+        filter_query_text: null,
+        filter_users: null,
+        filter_not_users: null,
+    };
+
+    const queryText = parsedQuery.getAllText();
+    if (queryText) {
+        filters.filter_query_text = queryText;
+    }
+
+    for (const option of parsedQuery.getConditionArray()) {
```

Comment:
This is actually pretty clean code and very easy to extend. I like it!

Reply:
Yeah, it's not bad! We could add a layer of abstraction to try to make it easier to add more fields and reuse this for questions/etc., but I'm pretty happy with how it looks now.

Reply:
Yeah, I think just hard-coding things like this is pretty good. No magic involved :-)